public/javascripts/modules/calculator.mjs

```diff
@@ -28,24 +28,97 @@ const addListeners = () => {
+      //   case "add":
+      //     return +n1 + +n2;
+      //   case "subtract":
+      //     return +n1 - +n2;
+      //   case "multiply":
+      //     return +n1 * +n2;
+      //   case "divide":
+      //     return +n1 / +n2;
+      //   // case "modulo": //use later
+      //   //   return +n1 % +n2;
+      //   default:
+      //     return result;
+      //
+      // }
+
       // If the calculator shows 0, we want to replace the calculator’s display with the textContent of the clicked key.
       //If the previousKeyType is an operator, we want to replace the displayed number with clicked number
       if (!action) {
-        if (displayedNum === 0 || previousKeyType === 'operator') {
+        if (
+            displayedNum === '0' ||
+            previousKeyType === 'operator' ||
+            previousKeyType === 'calculate'
+        ) {
           display.textContent = keyContent
         } else {
           display.textContent = displayedNum + keyContent
         }
+        calculator.dataset.previousKeyType = 'number'
       }
-      ;
       // If user hits any number after hitting a decimal key, the number should be appended on the display as well
       if (action === 'decimal') {
-        display.textContent = displayedNum + '.'
+        if (!displayedNum.includes('.')) {
+          display.textContent = displayedNum + '.'
+        } else if (
+            previousKeyType === 'operator' ||
+            previousKeyType === 'calculate'
+        ) {
+          display.textContent = '0.'
+        }
+        calculator.dataset.previousKeyType = 'decimal'
       }
-      ;
-
-      if (!action) {
-        console.log("You pressed a number key");
+      if (action !== 'clear') {
+        const clearButton = calculator.querySelector('[data-action=clear]')
+        clearButton.textContent = 'CE'
+      }
+      if (action === 'clear') {
+        if (key.textContent === 'AC') {
+          calculator.dataset.firstValue = ''
+          calculator.dataset.modValue = ''
+          calculator.dataset.previousKeyType = ''
+          calculator.dataset.operator = ''
+        } else {
+          key.textContent = 'AC'
+        }
+        display.textConent = 0
+        calculator.dataset.previousKeyType = 'clear'
+      }
+      if (action === 'calculate') {
+        let firstValue = calculator.dataset.firstValue;
+        const operator = calculator.dataset.operator;
+        let secondValue = displayedNum;
+        //console.log("you pressed the equal key")
+        if (firstValue) {
+          if (previousKeyType === 'calculate') {
+            firstValue = displayedNum
+            secondValue = calculator.dataset.modValue
+          }
+          display.textContent = calculate(firstValue, operator, secondValue)
+        }
+        // Set modValue attribute
+        calculator.dataset.modValue = secondValue
+        calculator.dataset.previousKeyType = 'calculate'
+      }
+      if (!displayedNum.includes('.')) {
```

Comment:
This if statement is running on every button press, but you only want to check for a `.` if a decimal button is pressed. Looks like you already have that logic above. I think you can just remove this line.